### PR TITLE
CEDS-1637 error messages

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

File: app/models/notifications/NotificationFrontendModel.scala [hunk 1]
```diff
@@ -28,8 +28,9 @@ final case class NotificationFrontendModel(
   ucrBlocks: Seq[UcrBlock],
   masterRoe: Option[String],
   masterSoe: Option[String],
+  crcCode: Option[String],
```
Comment (PatrykRudnicki) [inline app/models/notifications/NotificationFrontendModel.scala]: You changed the order of arguments. Can you confirm this? Both of the arguments has the same type `Option[String]` so in places where we can use it it'll contains wrong data.

File: app/models/viewmodels/NotificationPageSingleElementFactory.scala [hunk 2]
```diff
@@ -84,14 +83,30 @@ class NotificationPageSingleElementFactory @Inject()(decoder: Decoder) {
...
+    } yield content
+
+    val errorExplanation = (for {
+      code <- notification.errorCodes
+      errorCode <- decoder.errorCode(code) match {
+        case None =>
+          logger.warn(s"Received inventoryLinkingControlResponse with unknown error code: $code")
+          None
+        case knownCode => knownCode
+      }
```
Comment (PatrykRudnicki) [inline app/models/viewmodels/NotificationPageSingleElementFactory.scala]: This code return the same element. What do you think about something like this

```
def method(a: Option[String]): Option[String] =
  if(a.isEmpty) logger.warn(...)
  a
}
```

Of course with better naming or without function

```
errorCode <- {
  val code = decoder.errorCode(code)
  if(code.isEmpty) logger.warn(...)
  code
}
```

File: test/models/viewmodels/decoder/RoeCodeSpec.scala [hunk 3]
```diff
@@ -41,7 +40,7 @@ class RoeCodeSpec extends UnitSpec {
         PrelodgePrefix
       )
 
-      RoeCode.codes mustBe expectedCodes
+      RoeCode.codes mustEqual expectedCodes
```
Comment (PatrykRudnicki) [inline test/models/viewmodels/decoder/RoeCodeSpec.scala]: Why `mustEqual`?

File: app/models/viewmodels/NotificationPageSingleElementFactory.scala (hunk 2 above)
Comment (MaciejRewera) [reply app/models/viewmodels/NotificationPageSingleElementFactory.scala]: :+1:

File: app/models/notifications/NotificationFrontendModel.scala (hunk 1 above)
Comment (MaciejRewera) [reply app/models/notifications/NotificationFrontendModel.scala]: Double checked, run tests. Everything looks fine and working.

File: app/models/viewmodels/NotificationPageSingleElementFactory.scala (hunk 2 above)
Comment (MaciejRewera) [reply app/models/viewmodels/NotificationPageSingleElementFactory.scala]: Used you second solution.

File: test/models/viewmodels/decoder/RoeCodeSpec.scala (hunk 3 above)
Comment (MaciejRewera) [reply test/models/viewmodels/decoder/RoeCodeSpec.scala]: Changed back to `mustBe` in a separate commit. In this case there is no difference in functionality, but should compile slightly faster.

File: app/models/viewmodels/NotificationPageSingleElementFactory.scala [hunk 4]
```diff
@@ -84,14 +83,30 @@ class NotificationPageSingleElementFactory @Inject()(decoder: Decoder) {
     notification: NotificationFrontendModel
   )(implicit messages: Messages): NotificationsPageSingleElement = {
 
-    val content = notification.actionCode.map { code =>
-      s"<p>${messages(decoder.actionCode(code).map(_.contentKey).getOrElse(""))}</p>"
-    }
+    val actionCodeExplanation = for {
+      code <- notification.actionCode
+      actionCode <- decoder.actionCode(code)
+      content = s"<p>${messages(actionCode.contentKey)}</p>"
```
Comment (bambuchaAdm) [inline app/models/viewmodels/NotificationPageSingleElementFactory.scala]: why in loop

File: app/models/viewmodels/decoder/ErrorCode.scala [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.viewmodels.decoder
+
+sealed abstract class ErrorCode(override val code: String, override val status: String, override val contentKey: String)
```
Comment (bambuchaAdm) [inline app/models/viewmodels/decoder/ErrorCode.scala]: scaladoc

**Pull request merged**